### ISSUE TITLE
LOADING A TORCHSCRIPT MODEL IN C++ bug: c10::Error: isTensor() INTERNAL ASSERT FAILED

### ISSUE THREAD

**Issue by sjf18** [pytorch/pytorch#32039]
## 🐛 Bug

when i'm loading a torchscript model in c++,  bug:

libc++abi.dylib: terminating with uncaught exception of type c10::Error: isTensor() INTERNAL ASSERT FAILED at /Users/sjf/libtorch/include/ATen/core/ivalue_inl.h:86, please report a bug to PyTorch. Expected Tensor but got Tuple (toTensor at /Users/sjf/libtorch/include/ATen/core/ivalue_inl.h:86)
frame #0: c10::Error::Error(c10::SourceLocation, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) + 135 (0x104a289e7 in libc10.dylib)
frame #1: c10::IValue::toTensor() && + 289 (0x1049a2b81 in test)
frame #2: main + 651 (0x1049a1ddb in test)
frame #3: start + 1 (0x7fff666cf7fd in libdyld.dylib)
frame #4: 0x0 + 1 (0x1 in ???)
zsh: abort      ./test

## To Reproduce

1. Firstly, i have converted my own pytorch model to torch script via tracing.

2. Here is my cmakelists.txt
`cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
project(custom_ops)
find_package(Torch REQUIRED)
add_executable(test main.cpp)
target_link_libraries(test "${TORCH_LIBRARIES}") 
set_property(TARGET test PROPERTY CXX_STANDARD 11)`

3. Here is my cpp code
`#include <torch/script.h>
#include <iostream>
#include <memory>

int main(int argc, const char* argv[]) {
    torch::jit::script::Module module;
    try {
        module = torch::jit::load("/Users/alon/Downloads/bise_x39_102.pt");
    }
    catch (const c10::Error& e) {
        std::cerr << "error loading the model\n";
        return -1;
    }
    std::vector<torch::jit::IValue> inputs;
    inputs.push_back(torch::ones({1, 3, 384, 384}));
    at::Tensor output = module.forward(inputs).toTensor();
}`
 when i run it, it have bugs, i'm sure it has something wrong in ` at::Tensor output = module.forward(inputs).toTensor();`, but i have no idea what's wrong with it.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):1.3
 - OS (e.g., Linux):MACOSX 10.15.2
 - How you installed PyTorch (`conda`, `pip`, source):pip
 - Build command you used (if compiling from source):
 - Python version:3.7.4
 - CUDA/cuDNN version: None
 - GPU models and configuration: None
 - Any other relevant information:

**Comment by driazati** [comment]
From the error message `Expected Tensor but got Tuple` it sounds like your model returns a tuple. So instead of `.toTensor()` you should be doing `.toTuple()`. You can read more about the `torch::IValue` returned from `forward()` [here](https://pytorch.org/cppdocs/api/structc10_1_1_i_value.html#_CPPv4N3c106IValueE)

**Comment by xsacha** [comment]
Perhaps needs a better assert message. You can also check if it is a Tuple before converting, for code that handles multiple models.

**Comment by suo** [comment]
Agreed on the assert message, we're looking to clean up and better document the TorchScript C++ API in the very near future!

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
